mcp/gcp_cloudrun.tf

```diff
@@ -0,0 +1,112 @@
+
+//noinspection HILUnresolvedReference
+data "google_project" "default" {
+  project_id = lookup(local.cloudrun, "project_id", "")
+}
+
+data "google_container_registry_image" "default"{
+  name = lookup(local.cloudrun_specs, "image_name", "")
+  project = data.google_project.default.project_id
+}
+
+//noinspection HILUnresolvedReference
+resource "google_cloud_run_service" "self" {
+  count = local.cloudrun == {} ? 0 : 1
+  location = local.cloudrun.location_id
+  name     = local.cloudrun_specs.name
+  project  = data.google_project.default.project_id
+  template {
+    spec{
+      //noinspection HILUnresolvedReference
+      containers {
+        image = data.google_container_registry_image.default.image_url
+        //noinspection HILUnresolvedReference
+        dynamic "env"{
+          for_each = lookup(local.cloudrun_specs, "environment_vars", {})
+          content {
+            name  = env.key
+            value = env.value
+          }
+        }
+      }
+    }
+    //noinspection HILUnresolvedReference
+    metadata {
+      name        = length(local.cloudrun_specs.metadata) > 0 ? lookup(local.cloudrun_specs.metadata, "name", null): null
+      annotations = length(local.cloudrun_specs.metadata) > 0 ? lookup(local.cloudrun_specs.metadata, "annotations", null) : null
+    }
+  }
+  dynamic "traffic" {
+    for_each = local.cloudrun_traffic
+    //noinspection HILUnresolvedReference
+    content{
+      percent = lookup(traffic.value, "percent", null)
+      revision_name = lookup(traffic.value, "revision_name", null)
+      latest_revision = lookup(traffic.value, "latest_revision", null)
+    }
+  }
+
+}
+
+data "google_iam_policy" "noauth" {
+  binding {
+    role = "roles/run.invoker"
+    members = ["allUsers"]
+  }
+}
+
+data "google_iam_policy" "auth" {
+  //noinspection HILUnresolvedReference
+  binding {
+    role = "roles/${lookup(local.cloudrun_iam, "role", "" )}"
+    members = local.cloudrun_iam_members
+  }
+}
+
+//noinspection HILUnresolvedReference
+resource "google_cloud_run_service_iam_policy" "policy" {
+  count       = lookup(local.cloudrun_iam, "replace_policy", true) ? 1 : 0
```

Comment:
like what is mentioned in the [structure](https://github.com/mesoform/terraform-infrastructure-modules/tree/master/mcp#structure), there will often be considerations for this to be a collection of micro services. Look to use for_each and follow the pattern for GAE